jobfunnel/backend/scrapers/indeed.py

```diff
@@ -229,11 +229,42 @@ def _get_search_url(self, method: Optional[str] = 'get') -> str:
         TODO: use Enum for method instead of str.
         """
         if method == 'get':
-            return (
-                "https://www.indeed.{}/jobs?q={}&l={}%2C+{}&radius={}&"
-                "limit={}&filter={}{}".format(
+            search_url = "https://www.indeed.{}/jobs?q={}".format(
                     self.config.search_config.domain,
-                    self.query,
+                    self.query
+                )
+            # search countrywide supposing one is open to relocating within
+            # the country (includes remote)
+            if self.config.search_config.city is None and \
+                self.config.search_config.province_or_state is None and \
+                self.config.search_config.remote_within_country is False:
+                return search_url
+            # search remote within the country
+            if self.config.search_config.remote_within_country is True:
+                return (
+                        search_url + "&l=remote&"
+                        "limit={}&filter={}{}".format(
+                        self.config.search_config.province_or_state.upper(),
+                        self.max_results_per_page,
+                        int(self.config.search_config.return_similar_results),
+                        REMOTENESS_TO_QUERY[self.config.search_config.remoteness]
```

Comment:
I think it would make more sense for us to construct the URL in an additive way vs this if/else and return method. This way we can easily add or remove portions and update to using a URL-construction library (Which helps with constructing the URL string from tags).

Reply:
i.e. 
```
if search_config.province_or_state: 
    search_url += "...."
```